openshift-kube-apiserver/admission/customresourcevalidation/authentication/validate_authentication.go

```diff
@@ -121,14 +147,122 @@ func validateAuthenticationSpec(spec configv1.AuthenticationSpec) field.ErrorLis
+		errs = append(errs, err...)
+		warns = append(warns, warnings...)
+	}
+	// ----------------
+
+	return errs, warns
 }
 
 func validateAuthenticationStatus(status configv1.AuthenticationStatus) field.ErrorList {
 	return crvalidation.ValidateConfigMapReference(field.NewPath("status", "integratedOAuthMetadata"), status.IntegratedOAuthMetadata, false)
 }
+
+func validateOIDCProvider(path *field.Path, cel celStore, provider configv1.OIDCProvider) (field.ErrorList, []string) {
+	return validateClaimMappings(path, cel, provider.ClaimMappings)
+}
+
+func validateClaimMappings(path *field.Path, cel celStore, claimMappings configv1.TokenClaimMappings) (field.ErrorList, []string) {
+	path = path.Child("claimMappings")
+	errs := field.ErrorList{}
+	warns := []string{}
+
+	err, warn := validateUIDClaimMapping(path, cel, claimMappings.UID)
+	errs = append(errs, err...)
+	warns = append(warns, warn)
+
+	err, warnings := validateExtraClaimMapping(path, cel, claimMappings.Extra...)
+	errs = append(errs, err...)
+	warns = append(warns, warnings...)
+
+	return errs, warns
+}
+
+func validateUIDClaimMapping(path *field.Path, cel celStore, uid *configv1.TokenClaimOrExpressionMapping) (field.ErrorList, string) {
+	if uid == nil {
+		return nil, ""
+	}
+
+	var warn string
+	if uid.Expression != "" {
+		var err error
+		err, warn = validateCELExpression(cel, &authenticationcel.ClaimMappingExpression{
+			Expression: uid.Expression,
+		})
+		if err != nil {
+			return field.ErrorList{field.Invalid(path.Child("uid", "expression"), uid.Expression, err.Error())}, warn
+		}
+	}
+
+	return nil, warn
+}
+
+func validateExtraClaimMapping(path *field.Path, cel celStore, extras ...configv1.ExtraMapping) (field.ErrorList, []string) {
+	errs := field.ErrorList{}
+	warns := []string{}
+	for i, extra := range extras {
+		err, warn := validateExtra(path.Child("extra").Index(i), cel, extra)
+		errs = append(errs, err...)
+		warns = append(warns, warn)
+	}
+	return errs, warns
+}
+
+func validateExtra(path *field.Path, cel celStore, extra configv1.ExtraMapping) (field.ErrorList, string) {
+	err, warn := validateCELExpression(cel, &authenticationcel.ExtraMappingExpression{
+		Key:        extra.Key,
+		Expression: extra.ValueExpression,
+	})
+	if err != nil {
+		return field.ErrorList{field.Invalid(path.Child("valueExpression"), extra.ValueExpression, err.Error())}, warn
+	}
+
+	return nil, warn
+}
+
+func validateCELExpression(cel celStore, accessor authenticationcel.ExpressionAccessor) (error, string) {
+	warningString := ""
+	compiled := make(chan error, 1)
+
+	go func() {
+		if cel.knownCompilableExpressions.Has(accessor.GetExpression()) {
+			compiled <- nil
+			return
+		}
+
+		_, err := cel.compiler.CompileClaimsExpression(accessor)
+		if err != nil {
+			compiled <- err
+			return
+		}
+
+		cel.knownCompilableExpressions.Insert(accessor.GetExpression())
+		compiled <- nil
+	}()
+
+	slowCompile := time.After(time.Second)
+
+	for {
+		select {
+		case <-slowCompile:
```

Comment:
Nit: IMO introducing a variable for this hurts readability.

```suggestion
	for {
		select {
		case <-time.After(time.Second):
```

Reply:
Go playground to show using the variable causes a trigger once vs multiple triggers: https://go.dev/play/p/68dg9LfhpWv